### PR TITLE
GH-3197 - Fixes several errors regarding Google login

### PR DESCRIPTION
**Pull request by biafra23**
Handles #3183, #3191, #3197, #3199

Changes proposed in this pull request:
- Listen to notification send when CustomAuth.handle() is called via the ASWebAuthenticationSession completion closure
- Make self reference weak in OnboardingViewController
-

### CODE REVIEW

File: Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift [hunk 1]
```diff
@@ -22,29 +22,35 @@ class GoogleWeb3AuthLoginModel {
...
+                )
+                let tdsdk = CustomAuth(aggregateVerifierType: .singleLogin,
+                                       aggregateVerifier: App.configuration.web3auth.googleVerifier,
+                                       subVerifierDetails: [sub],
+                                       network: .CYAN
+                )
+                await MainActor.run(body: {
+                    authorizationComplete()
+                })
+                let data = try await tdsdk.triggerLogin(browserType: .asWebAuthSession)
```
Comment (biafra23) [inline Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift]: triggerLogin() is declared as throwing. But the print in the catch clause (line 52) is never reached.

File: Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift [hunk 2]
```diff
@@ -211,6 +210,13 @@ class CreateSafeFlow: UIFlow, ASAuthorizationControllerPresentationContextProvid
         } else if model.state == .error {
             LogService.shared.error("---> updateUI() called: state: \(model.state)")
         }
+
+        if model.state == .error {
```
Comment (MouazAlzahabi) [inline Multisig/UI/Safe Management/Create Safe/CreateSafeFlow.swift]: Why new "if" statement and not in the line above ?

File: Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift [hunk 3]
```diff
@@ -33,9 +28,6 @@ class GoogleWeb3AuthLoginModel {
                                    subVerifierDetails: [sub],
                                    network: .CYAN
             )
-            await MainActor.run(body: {
```
Comment (MouazAlzahabi) [inline Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift:36]: Are you sure this is not needed anymore?, then why it was there in the original place?

Comment (biafra23) [reply Multisig/UI/Safe Management/Create Safe/GoogleWeb3AuthLoginModel.swift:36]: I think it worked by accident. :-/

**Review by MouazAlzahabi — APPROVED** [review]

**Pull request merged**